src/System.Management.Automation/engine/SessionStateLocationAPIs.cs

```diff
@@ -272,10 +271,7 @@ internal PathInfo SetLocation(string path, CmdletProviderContext context, bool l
             PSDriveInfo previousWorkingDrive = CurrentDrive;
 
             // First check to see if the path is a home path
-            if (LocationGlobber.IsHomePath(path))
-            {
-                path = Globber.GetHomeRelativePath(path);
-            }
+            path = Globber.GetHomeRelativePath(path);
```

Comment:
`path` is used several times after this.
I'd like to review this is a separate PR and make sure we have adequate test coverage to ensure that we don't regress.

Reply:
GetHomeRelativePath() calls IsHomePath(path) internally so here we can remove the duplication call - no functional changes.

Reply:
Please explain in more detail you logic here.  If the path variable is changed more often, then there could be functional changes despite the internal implementation.

Reply:
Before the change we call GetHomeRelativePath() only if IsHomePath() returns true. If we look GetHomeRelativePath() the method returns original Path without change if  IsHomePath() returns false, otherwise it can calculate relative path. So first  IsHomePath() call can be removed.

Reply:
The logic in the function doesn't seem to match this explanation.  Please split the changes where you make this assumption into a new PR.

Reply:
In short the logic is
```c#
        internal string GetHomeRelativePath(string path)
        {
            string result = path;

            if (IsHomePath(path) && _sessionState.Drive.Current != null)
            {
                … // Calculate relative path
                result = path;
            }

            return result;
        }
```
If IsHomePath() returns false we return original path - we have the same logic before the change.